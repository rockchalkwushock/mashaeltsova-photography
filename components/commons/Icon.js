import PropTypes from 'prop-types'
import FontAwesome from 'react-fontawesome'

// Get current year from browser.
const year = new Date().getFullYear()
// Slick thinking!
const yearToDisplay = year === 2017 ? year : `2017 - ${year}`

// NOTE: She wants this left in English!
const Icon = ({ icon, size, url }) => {
  if (icon === 'copyright') {
    return (
      <a href={url}>
        <span>{`Designed by RCWS Development ${yearToDisplay}`}</span>
        <style jsx>{`
          @media (max-width: 321px) {
            span {
              font-size: 0.95em;
            }
          }
        `}</style>
      </a>
    )
  }
  return (
    <a href={url}>
      <FontAwesome name={icon} size={size} />
    </a>
  )
}

Icon.propTypes = {
  icon: PropTypes.string.isRequired,
  size: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}

export default Icon
