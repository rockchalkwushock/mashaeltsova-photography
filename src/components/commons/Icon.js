import React from 'react'
import styled from 'styled-components'
import fontawesome from '@fortawesome/fontawesome'
import FontAwesomeIcon from '@fortawesome/react-fontawesome'
import faBars from '@fortawesome/fontawesome-free-solid/faBars'
import faEnvelope from '@fortawesome/fontawesome-free-solid/faEnvelope'
import faFacebook from '@fortawesome/fontawesome-free-brands/faFacebook'
import faInstagram from '@fortawesome/fontawesome-free-brands/faInstagram'
import faVk from '@fortawesome/fontawesome-free-brands/faVk'

fontawesome.library.add(faBars, faFacebook, faEnvelope, faInstagram, faVk)

const UnstyledIcon = props => {
  // Removes error: 'Received "menu": `true` for non-boolean from fontawesome
  const rest = { ...props, menu: null }
  return <FontAwesomeIcon {...rest} />
}

export const StyledIcon = styled(UnstyledIcon)`
  margin-left: ${({ menu }) => (menu ? '10px' : 'unset')}
  margin-top: ${({ menu }) => (menu ? '10px' : 'unset')}
`

const Icon = props => <StyledIcon {...props} />

export default Icon
