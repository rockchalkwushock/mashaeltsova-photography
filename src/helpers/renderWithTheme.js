import React from 'react'
import { mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from '../utils/theme'

/**
 * @fileOverview
 *
 * HOC for testing components with `theme` prop being used.
 */

const renderWithTheme = Component =>
  mount(<ThemeProvider theme={theme}>{Component}</ThemeProvider>)

export default renderWithTheme
