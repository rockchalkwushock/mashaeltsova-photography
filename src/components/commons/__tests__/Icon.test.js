import React from 'react'
import { mount } from 'enzyme'

import Icon, { StyledIcon } from '../Icon'
import { renderWithTheme } from '../../../helpers'

describe('Component: <Icon />', () => {
  let wrapper
  let wrapper2
  beforeAll(() => {
    wrapper = mount(<Icon icon="bars" size="2x" />)
    wrapper2 = mount(<Icon icon="bars" menu size="2x" />)
  })
  test('should render markup as Icon', () => {
    expect(wrapper).toMatchSnapshot()
  })
  test('should render markup as MenuIcon', () => {
    expect(wrapper2).toMatchSnapshot()
  })
  test('Icon should have props', () => {
    expect(wrapper.props()).toHaveProperty('icon', 'bars')
    expect(wrapper.props()).toHaveProperty('size', '2x')
  })
  test('MenuIcon should have props', () => {
    expect(wrapper2.props()).toHaveProperty('menu', true)
  })
})

describe('Styled Component: <StyledIcon />', () => {
  let wrapper
  let wrapper2
  beforeEach(() => {
    wrapper = renderWithTheme(<StyledIcon icon="bars" size="2x" />)
    wrapper2 = renderWithTheme(<StyledIcon icon="bars" menu size="2x" />)
  })
  test('should render markup as Icon', () => {
    expect(wrapper).toMatchSnapshot()
  })
  test('should render markup as MenuIcon', () => {
    expect(wrapper2).toMatchSnapshot()
  })
  test('should have styles as Icon', () => {
    expect(wrapper.find('UnstyledIcon')).toHaveStyleRule('margin-left', 'unset')
    expect(wrapper.find('UnstyledIcon')).toHaveStyleRule('margin-top', 'unset')
  })
  test('should have styles as MenuIcon', () => {
    expect(wrapper2.find('UnstyledIcon')).toHaveStyleRule('margin-left', '10px')
    expect(wrapper2.find('UnstyledIcon')).toHaveStyleRule('margin-top', '10px')
  })
})
