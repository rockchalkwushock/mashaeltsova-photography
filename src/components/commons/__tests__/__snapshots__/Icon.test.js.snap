// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <Icon /> should render markup as Icon 1`] = `
.c0 {
  margin-left: unset;
  margin-top: unset;
}

<Icon
  icon="bars"
  size="2x"
>
  <Styled(UnstyledIcon)
    icon="bars"
    size="2x"
  >
    <UnstyledIcon
      className="c0"
      icon="bars"
      size="2x"
    >
      <FontAwesomeIcon$1
        border={false}
        className="c0"
        fixedWidth={false}
        flip={null}
        icon="bars"
        listItem={false}
        mask={null}
        menu={null}
        name=""
        pull={null}
        pulse={false}
        rotation={null}
        size="2x"
        spin={false}
        symbol={false}
        transform={null}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-bars fa-w-14 fa-2x c0"
          data-icon="bars"
          data-prefix="fas"
          menu={null}
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </FontAwesomeIcon$1>
    </UnstyledIcon>
  </Styled(UnstyledIcon)>
</Icon>
`;

exports[`Component: <Icon /> should render markup as MenuIcon 1`] = `
.c0 {
  margin-left: 10px;
  margin-top: 10px;
}

<Icon
  icon="bars"
  menu={true}
  size="2x"
>
  <Styled(UnstyledIcon)
    icon="bars"
    menu={true}
    size="2x"
  >
    <UnstyledIcon
      className="c0"
      icon="bars"
      menu={true}
      size="2x"
    >
      <FontAwesomeIcon$1
        border={false}
        className="c0"
        fixedWidth={false}
        flip={null}
        icon="bars"
        listItem={false}
        mask={null}
        menu={null}
        name=""
        pull={null}
        pulse={false}
        rotation={null}
        size="2x"
        spin={false}
        symbol={false}
        transform={null}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-bars fa-w-14 fa-2x c0"
          data-icon="bars"
          data-prefix="fas"
          menu={null}
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </FontAwesomeIcon$1>
    </UnstyledIcon>
  </Styled(UnstyledIcon)>
</Icon>
`;

exports[`Styled Component: <StyledIcon /> should render markup as Icon 1`] = `
.c0 {
  margin-left: unset;
  margin-top: unset;
}

<ThemeProvider
  theme={
    Object {
      "sizes": Object {
        "lg": "1000px",
        "md": "736px",
        "sm": "540px",
      },
    }
  }
>
  <Styled(UnstyledIcon)
    icon="bars"
    size="2x"
  >
    <UnstyledIcon
      className="c0"
      icon="bars"
      size="2x"
    >
      <FontAwesomeIcon$1
        border={false}
        className="c0"
        fixedWidth={false}
        flip={null}
        icon="bars"
        listItem={false}
        mask={null}
        menu={null}
        name=""
        pull={null}
        pulse={false}
        rotation={null}
        size="2x"
        spin={false}
        symbol={false}
        transform={null}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-bars fa-w-14 fa-2x c0"
          data-icon="bars"
          data-prefix="fas"
          menu={null}
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </FontAwesomeIcon$1>
    </UnstyledIcon>
  </Styled(UnstyledIcon)>
</ThemeProvider>
`;

exports[`Styled Component: <StyledIcon /> should render markup as MenuIcon 1`] = `
.c0 {
  margin-left: 10px;
  margin-top: 10px;
}

<ThemeProvider
  theme={
    Object {
      "sizes": Object {
        "lg": "1000px",
        "md": "736px",
        "sm": "540px",
      },
    }
  }
>
  <Styled(UnstyledIcon)
    icon="bars"
    menu={true}
    size="2x"
  >
    <UnstyledIcon
      className="c0"
      icon="bars"
      menu={true}
      size="2x"
    >
      <FontAwesomeIcon$1
        border={false}
        className="c0"
        fixedWidth={false}
        flip={null}
        icon="bars"
        listItem={false}
        mask={null}
        menu={null}
        name=""
        pull={null}
        pulse={false}
        rotation={null}
        size="2x"
        spin={false}
        symbol={false}
        transform={null}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-bars fa-w-14 fa-2x c0"
          data-icon="bars"
          data-prefix="fas"
          menu={null}
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </FontAwesomeIcon$1>
    </UnstyledIcon>
  </Styled(UnstyledIcon)>
</ThemeProvider>
`;
